### PR TITLE
Added check for empty query_name and where targetAddr may be empty

### PR DESCRIPTION
**Pull request by Conorbro**

### CODE REVIEW

File: dns.go [hunk 1]
```diff
@@ -109,9 +109,13 @@ func probeDNS(ctx context.Context, target string, module Module, registry *prome
 		probeDNSAdditionalRRSGauge.Set(float64(numAdditional))
 	}()
 
+	if module.DNS.QueryName == "" {
```
Comment (brian-brazil) [inline dns.go]: This should be part of the config validation, we shouldn't accept a config with this missing.

File: dns.go [hunk 2]
```diff
@@ -120,6 +124,9 @@ func probeDNS(ctx context.Context, target string, module Module, registry *prome
 		if port == "" {
 			port = "53"
 		}
+		if targetAddr == "" {
+			targetAddr = target
```
Comment (brian-brazil) [inline dns.go]: Combine this with the previous if, and check err instead

**Comment by Conorbro** [comment]
@brian-brazil

File: dns.go [hunk 3]
```diff
@@ -110,15 +110,16 @@ func probeDNS(ctx context.Context, target string, module Module, registry *prome
...
 	}
 	if module.DNS.TransportProtocol == "udp" || module.DNS.TransportProtocol == "tcp" {
-		targetAddr, port, _ := net.SplitHostPort(target)
-		if port == "" {
+		targetAddr, port, err := net.SplitHostPort(target)
+		if err != nil {
+			// Target only contains host so fallback to default port and set targetAddr as target.
 			port = "53"
+			targetAddr = target
+			log.Error(err)
```
Comment (brian-brazil) [inline dns.go]: This is an expected thing, so don't log it.

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**